fbw-a32nx/src/systems/instruments/src/EWD/NewPseudoFWC.ts

```diff
@@ -0,0 +1,2253 @@
+        this.allCurrentFailures.push(...allFailureKeys);
+
+        this.failuresLeft.length = 0;
+        this.failuresLeft.push(...failureKeysLeft);
+
+        this.failuresRight.length = 0;
+        this.failuresRight.push(...failureKeysRight);
+
+        if (tempFailureArrayLeft.length > 0) {
+            this.memoMessageLeft.set(orderedFailureArrayLeft);
+        }
+
+        for (const [, value] of Object.entries(this.ewdMessageMemos)) {
+            if (value.simVarIsActive.get() && !(value.memoInhibit()) && !value.flightPhaseInhib.some((e) => e === flightPhase)) {
+                const newCode: string[] = [];
+
+                const codeIndex = value.whichCodeToReturn().filter((e) => e !== null);
+                codeIndex.forEach((e: number) => {
+                    newCode.push(value.codesToReturn[e]);
+                });
+
+                if (value.side === 'LEFT' && !failLeft) {
+                    tempMemoArrayLeft = tempMemoArrayLeft.concat(newCode);
+                }
+                if (value.side === 'RIGHT') {
+                    const tempArrayRight = tempMemoArrayRight.filter((e) => !value.codesToReturn.includes(e));
+                    tempMemoArrayRight = tempArrayRight.concat(newCode);
+                }
+
+                if (value.sysPage > -1) {
+                    SimVar.SetSimVarValue('L:A32NX_ECAM_SFAIL', 'number', value.sysPage);
+                }
+            }
+        }
+
+        const mesgOrderLeft: string[] = [];
+        const mesgOrderRight: string[] = [];
+
+        for (const [, value] of Object.entries(this.ewdMessageMemos)) {
+            if (value.side === 'LEFT') {
+                mesgOrderLeft.push(...value.codesToReturn);
+            } else {
+                mesgOrderRight.push(...value.codesToReturn);
+            }
+        }
+
+        const orderedMemoArrayLeft = this.mapOrder(tempMemoArrayLeft, mesgOrderLeft);
+        let orderedMemoArrayRight = this.mapOrder(tempMemoArrayRight, mesgOrderRight);
+
+        if (!failLeft) {
+            this.memoMessageLeft.set(orderedMemoArrayLeft);
+
+            if (orderedFailureArrayRight.length === 0) {
+                this.masterCaution(0);
+                this.masterWarning(0);
+            }
+        }
+
+        if (leftFailureSystemCount + rightFailureSystemCount === 0) {
+            SimVar.SetSimVarValue('L:A32NX_ECAM_SFAIL', 'number', -1);
+        }
+
+        if (orderedFailureArrayRight.length > 0) {
+            // Right side failures need to be inserted between special lines
+            // and the rest of the memo
+            const specialLines = ['000014001', '000015001', '000035001', '000036001', '220001501', '220002101'];
+            const filteredMemo = orderedMemoArrayRight.filter((e) => !specialLines.includes(e));
+            const specLinesInMemo = orderedMemoArrayRight.filter((e) => specialLines.includes(e));
+            if (specLinesInMemo.length > 0) {
+                // @ts-expect-error
```

Comment:
Explanation would be nice on why this error is expected.

Reply:
This was something copied over from the old FWC, but I've found a fix.

The issue was that `orderedMemoArrayRight` was an untyped array, causing `filter()` to infer a `never[]` return type. TypeScript was mad about trying to merge a `never[]` and `[]` together. Adding the correct type hint to `orderedMemoArrayRight` has fixed it.